src/helper/proxy/index.ts

```diff
@@ -87,16 +111,7 @@ const buildRequestInitFromRequest = (
  * ```
  */
 export const proxy: ProxyFetch = async (input, proxyInit) => {
-  const { raw, ...requestInit } = proxyInit ?? {}
-
-  const req = new Request(
-    input,
-    // @ts-expect-error `headers` in `requestInit` is not compatible with HeadersInit
-    {
-      ...buildRequestInitFromRequest(raw),
-      ...requestInit,
-    }
-  )
```

Comment:
Re: #3919, my basic understanding of the spread operator is that new objects are created and all the keys are copied. This avoids that extra overhead